### PR TITLE
Time Loop Controller

### PR DESCRIPTION
**Pull request by emilyyyzh**
player stuck in time loop oh no

### CODE REVIEW

File: Assets/Scripts/PlayerMovementTime.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerMovementTime : MonoBehaviour
+{
+    // Player's rigidbody
+    Rigidbody rb;
+
+    // Player speed
+    [SerializeField] float speed = 3f;
```
Comment (wang-henry) [inline Assets/Scripts/PlayerMovementTime.cs:11]: I believe if we just make these public its by default serialized. See https://docs.unity3d.com/ScriptReference/SerializeField.html

Comment (wang-henry) [reply Assets/Scripts/PlayerMovementTime.cs:11]: Same for other code

File: Assets/Scripts/TimeLoopController.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        if (debugMode || resetting)
+        {
+            return;
+        }
+
+        if (timeCounter > 0) { 
+            // decrement time limit
+            timeCounter -= Time.deltaTime;
+
+            Debug.Log(timeCounter);
```
Comment (wang-henry) [inline Assets/Scripts/TimeLoopController.cs:33]: Should probably remove or the console will print on every frame.

**Review by wang-henry — APPROVED** [review]

File: Assets/Scripts/Persisting.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        if (instance == null)
+        {
+            instance = this;
+            DontDestroyOnLoad(gameObject);
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (emilyyyzh) [reply Assets/Scripts/Persisting.cs:22]: That method will duplicate non-destory objects on reload. The article fixes that issue using the same method I currently have.

File: Assets/Scripts/PlayerMovementTime.cs (hunk 1 above)
Comment (emilyyyzh) [reply Assets/Scripts/PlayerMovementTime.cs:11]: fixed

File: Assets/Scripts/TimeLoopController.cs (hunk 2 above)
Comment (emilyyyzh) [reply Assets/Scripts/TimeLoopController.cs]: fixed :)

**Pull request merged**